amundsen_application/static/js/components/BrowsePage/index.tsx

```diff
@@ -1,116 +1,22 @@
 import * as React from 'react';
 import * as DocumentTitle from 'react-document-title';
-import { connect } from 'react-redux';
-import { bindActionCreators } from 'redux';
 
-import './styles.scss';
-
-import AppConfig from 'config/config';
-import LoadingSpinner from 'components/common/LoadingSpinner';
-import TagInfo from 'components/Tags/TagInfo';
-import { Tag } from 'interfaces';
-
-import { GlobalState } from 'ducks/rootReducer';
-import { getAllTags } from 'ducks/allTags/reducer';
-import { GetAllTagsRequest } from 'ducks/allTags/types';
-
-export interface StateFromProps {
-  allTags: Tag[];
-  isLoading: boolean;
-}
-
-export interface DispatchFromProps {
-  getAllTags: () => GetAllTagsRequest;
-}
-
-interface BrowsePageState {
-  curatedTags: Tag[];
-  otherTags: Tag[];
-  isLoading: boolean;
-}
-
-export type BrowsePageProps = StateFromProps & DispatchFromProps;
-
-export class BrowsePage extends React.Component<BrowsePageProps, BrowsePageState> {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      curatedTags: [],
-      otherTags: [],
-      isLoading: this.props.isLoading,
-    };
-  }
-
-  static getDerivedStateFromProps(nextProps, prevState) {
-    const { allTags, isLoading } = nextProps;
-    if (isLoading) {
-      return {
-        ...prevState,
-        isLoading,
-      };
-    }
-
-    const curatedTagsList = AppConfig.browse.curatedTags;
-    const curatedTags = allTags.filter((tag) => curatedTagsList.indexOf(tag.tag_name) !== -1);
-
-    let otherTags = [];
-    if (AppConfig.browse.showAllTags) {
-      otherTags = allTags.filter((tag) => curatedTagsList.indexOf(tag.tag_name) === -1);
-    }
-    return { curatedTags, otherTags, isLoading };
-  }
-
-  componentDidMount() {
-    this.props.getAllTags();
-  }
-
-  generateTagInfo(tagArray: Tag[]) {
-    return tagArray.map((tag, index) =>
-      <TagInfo data={ tag } compact={ false } key={ index }/>)
-  }
+import BrowseTags from 'components/BrowseTags';
 
+export class BrowsePage extends React.Component {
   render() {
-    let innerContent;
-    if (this.state.isLoading) {
-      innerContent = <LoadingSpinner/>;
-    } else {
-      innerContent = (
+    return (
+      <DocumentTitle title="Browse - Amundsen">
         <div className="container">
           <div className="row">
             <div className="col-xs-12">
-              <h3 id="browse-header">Browse Tags</h3>
-              <hr className="header-hr"/>
-              <div id="browse-body" className="browse-body">
```

Comment:
I believe that part that needed to be broken out was only the portion that showed the actual tags. i.e. what was contained in `browse-body`. We want this page to still look exactly the same. Why? We shouldn't make adhoc changes to existing pieces of the product, only what is required for the feature.

That being said, I think it is correct for `BrowseTags` to still be the owner of the logic that connects to the store and fetches the tags.

Thoughts?

Reply:
Yeah that would make sense to just make the component the list of tags. The reason I didn't go down that path was I wanted to keep our components on the home page consistent - both bookmarks and populartables have titles as part of their respective components. Either way we go we should make it consistent across the board.

Reply:
That would be a case for having an existing `BrowsePage`, the nested `BrowseTags` component connected to the redux store -- which by the way would be a common component now -- and if you want the title as part of the component on the landing page then some third component. 

The three components may seem overkill since two of them will be really simple -- but I think that's a symptom of us having a whole page (`BrowsePage`) that is just one component in the first place. If `BrowsePage` ever gets more content or a different layout, we will be glad we made the 3 components.

Reply:
`PopularTables` fetches and connects the list of table resources. It also creates the title and places a `ResourceList` which takes care of displaying elements and pagination. For these components, there is no analogous full page component that displays only `PopularTables. Instead the `HomePage` is what houses and displays them.

If we're using `PopularTables/MyBookmarks` as an example, then...

- A `TagsList` component that would be responsible for displaying tags similarly to `ResourceList`. Instead of pagination, maybe it would have `[ + more]` `[ - less]` buttons (as per Philippe's mocks).
- A `BrowseTags` component would fetch and connect tags to the `TagsList`, perhaps with some logic about deciding which tags to show or hide. Maybe it has options to show top 20 tags, then click to show the rest.
- A `BrowsePage` would be like the `HomePage` that puts items together. Right now it only has tags, but this could be expanded to have other sections. Maybe there is an overall page title "Browse" and then a sub-section with "Tags", but that can all come once we have more things to show here. To be completely honest, the entire `BrowsePage` is sort of useless once the tags get added to the `HomePage`.